File: outputs/monster_range_skill_sync_20260713/update_workbook.mjs
import fs from "node:fs/promises";
import { FileBlob, SpreadsheetFile } from "@oai/artifact-tool";

const workbookPath = "D:/GameDev/ProjectDG/docs/DefenseGame_Balance_Skill_Summary.xlsx";
const input = await FileBlob.load(workbookPath);
const workbook = await SpreadsheetFile.importXlsx(input);
const result = await workbook.inspect({
  kind: "sheet,table",
  include: "id,name",
  maxChars: 12000,
  tableMaxRows: 8,
  tableMaxCols: 12,
});
console.log(result.ndjson);

File: outputs/monster_range_skill_sync_20260713/update_workbook.py
from pathlib import Path
import json

from openpyxl import load_workbook


WORKBOOK_PATH = Path(r"D:\GameDev\ProjectDG\docs\DefenseGame_Balance_Skill_Summary.xlsx")


def main() -> None:
    workbook = load_workbook(WORKBOOK_PATH, data_only=False)
    summary = []
    for sheet in workbook.worksheets:
        summary.append({
            "name": sheet.title,
            "max_row": sheet.max_row,
            "max_column": sheet.max_column,
        })
    print(json.dumps(summary, ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()
